cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Replica.java

```diff
@@ -140,6 +142,38 @@ void setBroker(Broker broker) {
     _broker = broker;
   }
 
+  /**
+   * Set disk that the replica resides in.
+   *
+   * @param disk Disk that the replica resides in.
+   */
+  void setDisk(Disk disk) {
+    _currentDisk = disk;
+  }
+
+  /**
+   * Set disk that the replica originally resides in.
+   *
+   * @param originalDisk Disk that the replica originally resides in.
+   */
+  void setOriginalDisk(Disk originalDisk) {
```

Comment:
The issue is because replica populate `_ originalDisk` optionally(only if [populateDiskInfo=true](https://github.com/linkedin/cruise-control/blob/ceed1c2ba7862360ab9a5f6f5c76ff6373f65aa3/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java#L540)) so it is done after the object get created.

I think it is a good idea to make sure _originalDisk is not get reassigned, I make it throw an Exception in that case.